src/lazy/text/buffer.rs

```diff
@@ -1009,7 +1060,7 @@ impl<'data> TextBufferView<'data> {
     /// Returns a matched buffer and a boolean indicating whether any escaped characters were
     /// found in the short string.
     pub(crate) fn match_short_string_body(self) -> IonParseResult<'data, (Self, bool)> {
-        Self::match_text_until_unescaped(self, b'\"')
+        Self::match_text_until_unescaped(self, b'\"', false)
```

Comment:
🗺️ Callers to this function must now specify whether unescaped newlines are permitted (long-form string literals) or not (short-form string literals).